elastic/kibana
# [Fleet] added task to check result of bulk action, saving error action results

## Summary

Follow up for https://github.com/elastic/kibana/issues/141567

### 1.
Improved bulk action batching implementation to add a scheduled task that checks that the action was executed successfully. This runs even if kibana is restarted.

To verify:
- create more than 10k agents ([create_agents](https://github.com/elastic/kibana/blob/main/x-pack/plugins/fleet/scripts/create_agents/index.js) script can be used)
- bulk action them, and shortly after the action started executing, stop and restart kibana
- verify that after restart, the action is completed (the kibana part of writing docs to `.fleet-actions` index)
- on successful action completion, the check task is removed

### 2.
Added logic in bulk actions to save action results with error for those agents that failed validation and are not sent to Fleet Server.

To verify:
- select 2 agents with different agent policies
- reassign the to one of the agent policies
- see in Agent activity that the action shows up as failed, indicating that 1 agent failed
- similarly unenroll and upgrade action should report failures coming from validation

Adding more tests in this pr.

### Checklist
